Allow VP without verifiableCredential or proof

clehner opened the pull request:
This addresses #112 to allow a `VerifiablePresentation` without the `verifiableCredential` property, for the use case of proving control over a DID as in [vp-request-spec](https://w3c-ccg.github.io/vp-request-spec/). I also removed the requirement for the `proof` property, as that property is similarly mentioned in the VC Data Model with "If present," implying optionality (https://w3c.github.io/vc-data-model/#presentations-0) - except for ZKPs where it is required (https://w3c.github.io/vc-data-model/#zero-knowledge-proofs).

I renamed the positive test case for a VP containing `verifiableCredential` and `proof` to say "MAY" instead of "MUST". "MAY" isn't used elsewhere in the tests, but I thought it would be useful to express that the credential may have this property (which the implementation must support). Since this test case is the same but only renamed, I considered updating the implementation reports JSON with the renaming, so that their info would still appear in the generated implementation report HTML file. But I left them as-is so they remain true to what the implementors submitted. This particular row in the generated implementation report then disappears until implementation reports are updated.

The negative case for missing `verifiableCredential` is changed to a positive case ("MAY omit `verifiableCredential`"). This means previous implementations passing this test will now fail it, and vice versa. But it will be useful to know which ones support this for the use cases in question.

A possible use case for omitting the proof property is in <https://github.com/w3c-ccg/vc-http-api/issues/88>, but I did not add any positive test case for omitting the proof property.

To replace the negative test case for missing proof property, I added a negative test case for a proof with missing `type` property, as that is a required property: https://w3c.github.io/vc-data-model/#proofs-signatures

Fixes #112

OR13 reviewed (approved):
yes please, see also https://github.com/w3c-ccg/vc-http-api/pull/95#issuecomment-777603045

msporny commented:
Merge conflicts.

I'd also like a few people to run these tests against their implementations to make sure they agree with the changes.

clehner commented:
Rebased to fix merge conflicts. I'm also open to other ways to make this work.

OR13 commented on implementations/generate.js:
```diff
@@ -43,7 +43,12 @@ const deprecatedTests = [
+  'Terms of Use (optional) MUST support prohibiting 3rd Party Correlation',
+  'Basic Documents Presentations MUST include `verifiableCredential` and `proof`',
+  'Basic Documents Presentations MUST include `verifiableCredential` and `proof` (negative - missing `verifiableCredential`)',
+  'Basic Documents Presentations MUST include `verifiableCredential` and `proof` (negative - missing `proof`)',
```
a presentation without a proof is valid according to the spec, this language does not feel precise enough for me.

clehner replied:
This is the list of deprecated tests, to which this PR adds 5. This list is used to remove the named tests from showing up in the generated `implementations/index.html`, even though they still may be present in some of the JSON reports.
The rows for the 5 deprecated tests are also removed from `implementations/index.html` in this PR.

OR13 commented on test/vc-data-model-1.0/10-basic.js:
```diff
@@ -218,22 +218,22 @@ describe('Basic Documents', function() {
     });
 
-    it('MUST include `verifiableCredential` and `proof` (negative - missing `verifiableCredential`)', async function() {
+    it('`proof`, if present, MUST include `type` property (negative)', async function() {
```
I see thee test here... I know the intent, just feels like the cases might be described in clearer langauge.

clehner replied:
Like this?
```
```

clehner replied:
I'm open to suggestions. Would it help to change it to start with a VERB, for consistency?
```
MAY include `proof`, which if present, MUST include `type` property (negative)
```
The part of the specification that this is intended to correspond to, [§ 4.7 Proofs (Signatures)](https://w3c.github.io/vc-data-model/#proofs-signatures), says:
> The specific method used for an embedded proof MUST be included using the type property

OR13 reviewed (approved):
minor nits, but looks good

msporny commented:
> Rebased to fix merge conflicts. I'm also open to other ways to make this work.

Conflicts are back :(

clehner committed:
Allow VP without verifiableCredential or proof

- Change MUST to MAY for positive test
- Change negative case to positive for MAY omit proof
- Test for proof type property

clehner commented:
Rebased again. The conflicts were in `implementations/index.html`; fixed using `node implementations/generate.js`

msporny commented:
@brentzundel -- can you please check the changes in this PR... it modifies the ZKP tests in a way that I think is correct, but want to double check with you to make sure I'm not missing something.
